IdentityCore/src/configuration/MSIDWebOpenBrowserAdditionalParameters.h

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface MSIDWebOpenBrowserAdditionalParameters : NSObject
+
+@property (atomic) NSMutableDictionary<NSString *, NSString *> *queryParameters;
+
++ (instancetype _Nonnull)sharedInstance;
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+- (void)addQueryParameterForKey:(NSString *)key value:(NSString *)value;
```

Comment:
Can we use methods present in NSURL+MSIDExtensions.h instead of creating a singleton to add query parameters? I think the recommendation was to use extensions/categories over singletons where ever possible

Reply:
This class is not just to add query string parameters to a NSURL, it will store data when SSO Extension configuration is retrieved (in broker side), and then use it to add the parameters stored when opening the browser (in common core side). 

For more context, this code goes with: https://github.com/AzureAD/WorkplaceJoin-for-iOS/pull/510